### ISSUE TITLE
COCO/ImageNet Implementation

### ISSUE THREAD

**Issue by matheus-hertzog-deel** [szagoruyko/wide-residual-networks#59]
In your wide-resnet.lua, line 95:
"--one conv at the beginning (spatial size: 32x32)"
I can see your model has input size of 32x32 (Cifar10/100 size) and later on performs an avg_pooling on an 8x8 input. 
How do you train your models on larger input datasets such as COCO and ImageNet? Do you just modify the avg pooling layer and the first convolutional layer?

**Comment by xwuShirley** [comment]
HI, @Matheusih  Have you figure out imagenet version of wide-resnet?

**Comment by matheus-hertzog-deel** [comment]
@xwuShirley sadly, no. The author does provide the model object, however, [here](https://github.com/szagoruyko/functional-zoo/blob/master/wide-resnet-50-2-export.ipynb)
I tried loading and testing it on the tiny-imagenet dataset but I couldn't get anything close to a desirable performance.

**Comment by xwuShirley** [comment]
@Matheusih  Have you tried input size with 224x224? would it work?

**Comment by szagoruyko** [comment]
see https://github.com/szagoruyko/wide-residual-networks/tree/master/pretrained for pretrained ImageNet WRN-50-2.

Also, support for WRN was added in torchvision via https://github.com/pytorch/vision/pull/822

To make BasicBlock WRN-18-2 do:

```python
from torchvision.models.resnet import Bottleneck, ResNet, BasicBlock
model = ResNet(BasicBlock, [2,2,2,2], width_per_group=64 * 2)
```

To make Bottleneck WRN-50-2 do:

```python
model = ResNet(Bottleneck, [3,4,6,3], width_per_group=64 * 2)
```

(this needs master torchvision)

Also, I plan adding pretrained WRN-18-2, WRN-34-2 and WRN-50-2.

*Update*: removed WRNBottleneck with modified expansion

**Comment by xwuShirley** [comment]
@szagoruyko Thank you so much for answering my question! My plan is actually to reproduce your result WRN-50-2 from imagenet as a baseline in pytorch with 8GPU. It would be great if you can provide some more information on how you train the network - some hyper-parameter setting.

1. What is the batch size for 8 GPU so that the training is stable? (assuming the random crop input is 224x224). As we know, in resnet50/101, 32image/gpu is good. But we perhaps need 16 image/gpu for  WRN-50-2? 
2. What is the learning rate schedule for you to get 21.9 in your paper table 8 (decay 30 epoch by a factor of 10)?  do you use SGD with both momentum and  Nesterov? What is weight-decay?

**Comment by xwuShirley** [comment]
@szagoruyko  Should the  [2,2,2,2] in the model of WRN-50-2 be  [3, 4, 6, 3] same as resnet50, the following?
`model = ResNet(WRNBottleneck, [3, 4, 6, 3], width_per_group=64 * 2)`
similarly, the resnet-34-2 like following? 
`ResNet(BasicBlock, [3, 4, 6, 3], width_per_group=64 * 2)`
very appreciate your help.

**Comment by szagoruyko** [comment]
@xwuShirley yes you're right, it is `[3, 4, 6, 4]`, fixed it in my post. default parameters for ResNet-50 from lua fb.resnet.torch or NVIDIA/apex should do for training.

**Comment by szagoruyko** [comment]
there was a fix in https://github.com/pytorch/vision/pull/852 so that it is no longer needed to make new Bottleneck class with modified expansion.

**Comment by szagoruyko** [comment]
pretrained models are now in pytorch hub, see https://pytorch.org/hub/pytorch_vision_wide_resnet/
